sklearn/utils/testing.py

```diff
@@ -625,8 +625,9 @@ def run_test(*args, **kwargs):
             import matplotlib
             matplotlib.use('Agg', warn=False)
             # this fails if no $DISPLAY specified
-            matplotlib.pylab.figure()
-        except:
+            import matplotlib.pyplot as plt
+            plt.figure()
+        except ImportError:
```

Comment:
Not sure I understand fully your point so I am going to try to reformulate mine in order to try to clarify ;-).

I reckon it's ok for now to make the assumption that matplotlib.pyplot has not been imported prior to `if_matplotlib` being called.

Maybe I am missing something but I just don't see what the `plt.figure()` line is here for (I do get why plt.figure() can fail in an environment without X as the snippet above illustrates). You could even argue that the `import matplotlib.pyplot as plt` is not necessary and could be removed. The only effect this has is a "soft" (because plt.switch_backend still has a running chance of being able to change the backend) freeze on the matplotlib backend.